include/arch/arm/arch.h

```diff
@@ -42,9 +42,98 @@ extern "C" {
 #include <arch/arm/cortex_m/sys_io.h>
 #include <arch/arm/cortex_m/nmi.h>
 #endif
+#if defined(CONFIG_MPU_STACK_GUARD)
+
```

Comment:
Stacks can really live anywhere that isn't in application memory space.
For stacks that are not defined as struct members, on x86 I put them in a special '.stacks' section which ends up in the kernel's "noinit".